docs/class-definitions.md

```diff
@@ -0,0 +1,73 @@
+## Class Definitions
+
+**Contents**
+
+- [Spaces](#spaces)
+- [SpaceOptions](#spaceoptions)
+- [Space](#space)
+- [SpaceMember](#spacemember)
+
+### Spaces
+
+#### constructor(ably)
+
+Create a new instance of the Spaces library.
+This requires a promise instance of the Ably library to be passed in.
+Refer to the [Ably-JS Documentation](https://github.com/ably/ably-js#introduction) for information on setting up a realtime promise client.
+
+| Property | Type                 |
+|----------|----------------------|
+| ably     | Ably.RealtimePromise |
+
+
+#### get(name, options?)
+Get or create a space. Options may only be provided if a space does not yet exist.
+
+| Property | Type                                           |
+|----------|------------------------------------------------|
+| name     | string                                         |
+| options  | [SpaceOptions](#spaceoptions) &#124; undefined |
+
+### SpaceOptions
+
+Used to configure a Space object on creation.
+
+#### offlineTimeout
+Number of milliseconds after a user loses connection or closes their browser window to wait before their [SpaceMember](#spacemember) object is removed from the members list.
+
+Default is 5000ms.
+
+### Space
+An instance of a Space created using [spaces.get](#getname-options).
+
+#### enter(profileData?)
+Enter this space with optional profile data and notifies other clients in the space via the [membersUpdate](#membersupdate) event.
+This data can be an arbitrary JSON-serializable object which will be attached to the member object and delivered to other members of the space.
+
+| Property    | Type   |
+|-------------|--------|
+| profileData | Object |
+
+
+#### leave()
+
+Leave the space. This removes you from the space and notifies other space members.
+
+#### on(event, callback)
+Used for subscribing to realtime events within the space. Currently, only one event is supported:
+
+##### membersUpdate
+Fires when a member is enters or leaves the space. This returns an array of [SpaceMember](#spacemember) objects.
```

Comment:
```suggestion
Fires when a member enters or leaves the space. This returns an array of [SpaceMember](#spacemember) objects.
```

Reply:
Also, this isn't a method call, so "this returns an array" doesn't seem appropriate